lib/model/model.go

```diff
@@ -1138,17 +1184,12 @@ func (m *model) handleIndex(conn protocol.Connection, folder string, fs []protoc
 	}
 
 	m.pmut.RLock()
-	indexHandler, ok := m.indexHandlers.Get(deviceID)
+	indexHandler, ok := m.getIndexHandlerPRLocked(conn)
 	m.pmut.RUnlock()
 	if !ok {
-		// This should be impossible, as an index handler always exists for an
-		// open connection, and this method can't be called on a closed
-		// connection
-		m.evLogger.Log(events.Failure, "index sender does not exist for connection on which indexes were received")
-		l.Debugf("%v for folder (ID %q) sent from device %q: missing index handler", op, folder, deviceID)
-		return fmt.Errorf("index handler missing: %s", folder)
+		l.Infof("%v for folder %s sent from %s (%s), but no index handler is registered for this connection.", op, folder, deviceID.Short(), conn)
+		return fmt.Errorf("%s: %w", folder, ErrFolderNotRunning)
```

Comment:
Isn't this still a failure, as in should never happen? The remote should only send us an index after successful cluster config exchange, and that only happens during connection promotion where the index handler is also added.

Reply:
I believe so; I don't remember why I changed commentary etc, reverting.